scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/AzureBlobDownloader.java

```diff
@@ -0,0 +1,221 @@
+import com.microsoft.azure.storage.blob.ContainerURL;
+import com.microsoft.azure.storage.blob.ListBlobsOptions;
+import com.microsoft.azure.storage.blob.models.BlobItem;
+import com.microsoft.azure.storage.blob.models.ContainerListBlobFlatSegmentResponse;
+import com.microsoft.rest.v2.util.FlowableUtil;
+
+import io.reactivex.Single;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 13/09/2018
+ */
+public class AzureBlobDownloader extends JavaExecutable {
+
+    private String outputPath;
+
+    private String containerName;
+
+    Optional<String> optionalBlobName;
+
+    private String accountName;
+
+    private String accountKey;
+
+    private static final String OUTPUT_PATH = "outputPath";
+
+    private static final String CONTAINER_NAME = "containerName";
+
+    private static final String BLOB_NAME = "blobName";
+
+    private ContainerURL containerURL;
+
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+
+        containerName = (String) Optional.ofNullable(args.get(CONTAINER_NAME))
+                                         .filter(container -> !container.toString().isEmpty())
+                                         .orElseThrow(() -> new IllegalArgumentException("You have to specify a container name. Empty value is not allowed."));
+
+        optionalBlobName = Optional.ofNullable(args.get(BLOB_NAME).toString()).filter(name -> !name.isEmpty());
+
+        //Default value is getLocalSpace() because it will always be writable and moreover can be used to transfer files to another data space (global, user)
+        outputPath = (String) Optional.ofNullable(args.get(OUTPUT_PATH))
+                                      .filter(output -> !output.toString().isEmpty())
+                                      .orElse(getLocalSpace());
+
+        // Retrieve the credentials
+        accountName = getThirdPartyCredential("ACCOUNT_NAME");
+        accountKey = getThirdPartyCredential("ACCOUNT_KEY");
+        if (accountName == null || accountKey == null) {
+            throw new IllegalArgumentException("You first need to add your account name and account key (ACCOUNT_NAME, ACCOUNT_KEY) to the third party credentials.");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException {
+
+        File file = new File(outputPath);
+        containerURL = AzureStorageConnectorUtils.createContainerURL(accountName, accountKey, containerName);
+        //download a single blob
+        if (optionalBlobName.isPresent()) {
+            //check weather or not the outputPath is a folder path or a file path
+            if (outputPath.lastIndexOf('/') == outputPath.length() - 1) {
+                createDirIfNotExists(file);
+                String azureBlobLocalRelativePath = Paths.get(outputPath,
+                                                              Paths.get(optionalBlobName.get())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "optionalBlobName.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)